wordpress-mobile/WordPress-iOS
# Update feature flag platform name

Fixes #20222

Updates the platform name we send to the feature flag endpoint. This PR relies on https://github.com/wordpress-mobile/WordPressKit-iOS/pull/582.

### To test

Do a quick smoke test of a change that relies on remote feature flags to ensure nothing is broken:
1. Log into a self-hosted site
2. Expect no Reader or Notifictions tab 

The removal of those features is controlled by the `jp_removal_self_hosted` remote feature flag. With this change, the app will fetch from https://public-api.wordpress.com/wpcom/v2/mobile/feature-flags?platform=ios&marketing_version=21.8 (note the `ios`).

## Regression Notes
1. Potential unintended areas of impact

This change could break feature flags, but as noted in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/582#pullrequestreview-1321137637, we haven't used the `platform` parameter yet, so no harm is done updating it to match the docs.

5. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing as described above.

6. What automated tests I added (or what prevented me from doing so)

I don't think an automated test for this is warranted.

PR submission checklist:
